src/_guides/openapi/specification/v3.1/the-perfect-modern-openapi-workflow.md

```diff
@@ -0,0 +1,706 @@
+`main` branch, but will tag versions as appropriate, and with a bit of other
+setup these SDKs can be pushed directly to package managers like NPM, PyPI,
+Packagist, NuGet, and Maven.
+
+Once this is done, you can update API documentation on Bump.sh to include these
+SDKs in the code examples, instead of the default of showing curl CLI examples,
+or rudimentary code samples like using `fetch()` or other low-level HTTP code.
+
+![The Train Travel API documentation example on Bump.sh with a TypeScript SDK generated by Speakeasy showing in the documentation.](/images/guides/the-perfect-modern-openapi-workflow/bump-sdks.png)
+
+*Learn more about [SDK generation with Speakeasy](_guides/bump-sh-tutorials/generate-sdks-with-speakeasy.md).*
+
+### API Catalog
+
+Another key part of API governance is discoverability, which usually takes the
+form of "API Catalogs". Infinite awkward solutions have been employed in the
+past. API teams have bodged together infinite awkward solutions with miles of
+duct-tape, but Bump.sh makes this easy with [Bump.sh
+Hubs](https://bump.sh/api-catalog).
+
+Each API that has been added to Bump.sh's hosted documentation can be grouped
+together into various Hubs, which could be for "Public" and "Internal" usage.
+Alternatively, Hubs could be used to group APIs by team or department. 
+
+![A list of APIs, descriptions, and last updated times, for a sample Hub on Bump.sh](/images/guides/the-perfect-modern-openapi-workflow/bump-hubs.png)
+
+*Learn more about [API discovery with Bump.sh Hubs](_guides/bump-sh-tutorials/api-discovery-using-bump-sh-hubs.md).*
+
+### Try it Now
+
+Seeing how an API works is the first step in an API consumers journey to using
+the API, and the second step is making some test requests to get a feel for how
+it works. Some people like to do this with code, so code samples will be a good
+start for them, especially if you have an SDL. Other people like to do this with
+interactive HTTP clients, like [Postman](https://postman.com/).
+
+Postman is able to import OpenAPI and create an API collection from it. You can
+manually do this step via the Postman interface, then pop a [Run in
+Postman](https://learning.postman.com/docs/publishing-your-api/run-in-postman/creating-run-button/)
+but into the Bump.sh API documentation to help API consumers find and use that
+collection in their Postman application.
+
+Keeping it up-to-date as the API evolves via [Postman's Git
+syncing](https://learning.postman.com/docs/designing-and-developing-your-api/versioning-an-api/versioning-an-api-overview/)
+can be a lot more involved, and require an more expensive plan, but you can
+cobble a solution together using the Postman Pro API if you are feeling up for
+it. 
+
+To avoid spending the time or money, you can let API consumers experiment with
+an API directly from the Bump.sh API documentation using the built-in API
+Explorer.
+
+![](/images/guides/the-perfect-modern-openapi-workflow/api-explorer-request.png)
+
```

Comment:
Very cool section. Should we add some recommendations on proper/convenient listing of servers in the OpenAPI definition file? Possibly some recommendations on having, or not, sandbox servers, or Mocks (hello again Microcks) whenever users should not hit the production servers from the docs?

Reply:
Great idea, I've done that but it could do with a screenshot. Can you enable the API Explorer for https://bump.sh/bump-examples/doc/train-travel-api/ so I can show it off?